failed to compile `clippy v0.0.111`

<!--
Hi there! Whether you've come to make a suggestion for a new lint, an improvement to an existing lint or to report a bug or a false positive in Clippy, you've come to the right place.

If you want to report that Clippy does not compile, please be sure to be using the *latest version* of *Rust nightly*! Compiler plugins are highly unstable and will only work with a nightly Rust for now. If you are but still have a problem, please let us now!

Thank you for using Clippy!

Write your comment below this line: -->
```
   Compiling clippy_lints v0.0.111
error[E0023]: this pattern has 2 fields, but the corresponding tuple variant has 3 fields
   --> /home/pawel/.cargo/registry/src/github.com-1ecc6299db9ec823/clippy_lints-0.0.111/src/utils/hir.rs:163:15
    |
163 |             (&PatKind::Range(ref ls, ref le), &PatKind::Range(ref rs, ref re)) => {
    |               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected 3 fields, found 2

error[E0023]: this pattern has 2 fields, but the corresponding tuple variant has 3 fields
   --> /home/pawel/.cargo/registry/src/github.com-1ecc6299db9ec823/clippy_lints-0.0.111/src/utils/hir.rs:163:48
    |
163 |             (&PatKind::Range(ref ls, ref le), &PatKind::Range(ref rs, ref re)) => {
    |                                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected 3 fields, found 2

error[E0023]: this pattern has 2 fields, but the corresponding tuple variant has 3 fields
   --> /home/pawel/.cargo/registry/src/github.com-1ecc6299db9ec823/clippy_lints-0.0.111/src/utils/inspector.rs:478:9
    |
478 |         hir::PatKind::Range(ref l, ref r) => {
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected 3 fields, found 2

error[E0023]: this pattern has 2 fields, but the corresponding tuple variant has 3 fields
   --> /home/pawel/.cargo/registry/src/github.com-1ecc6299db9ec823/clippy_lints-0.0.111/src/matches.rs:364:25
    |
363 |                       if_let_chain! {[
    |  _____________________- starting here...
364 | |                     let PatKind::Range(ref lhs, ref rhs) = pat.node,
    | |                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected 3 fields, found 2
365 | |                     let Ok(lhs) = constcx.eval(lhs, ExprTypeChecked),
366 | |                     let Ok(rhs) = constcx.eval(rhs, ExprTypeChecked)
367 | |                 ], {
368 | |                     return Some(SpannedRange { span: pat.span, node: (lhs, rhs) });
369 | |                 }}
    | |__________________- ...ending here: in this macro invocation

error: aborting due to 4 previous errors

error: failed to compile `clippy v0.0.111`, intermediate artifacts can be found at `/tmp/user/1000/cargo-install.IpUIDN9mFMZx`

Caused by:
  Could not compile `clippy_lints`.

To learn more, run the command again with --verbose.
```

* cargo 0.17.0-nightly (67e4ef1 2017-01-25)
* rustc 1.16.0-nightly (df8debf6d 2017-01-25)
* Linux i7 4.8.17-xanmod19 #1 SMP Mon Jan 9 23:03:38 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux
* Ubuntu 16.10

--- a/clippy_lints/src/lib.rs
+++ b/clippy_lints/src/lib.rs
@@ -14,7 +14,6 @@
 #![allow(indexing_slicing, shadow_reuse, unknown_lints, missing_docs_in_private_items)]
 #![allow(needless_lifetimes)]
 
-#[macro_use]
 extern crate syntax;
 #[macro_use]
 extern crate rustc;

--- a/clippy_lints/src/matches.rs
+++ b/clippy_lints/src/matches.rs
@@ -361,7 +361,7 @@ fn all_ranges(cx: &LateContext, arms: &[Arm]) -> Vec<SpannedRange<ConstVal>> {
                 }
                 .filter_map(|pat| {
                     if_let_chain! {[
-                    let PatKind::Range(ref lhs, ref rhs) = pat.node,
+                    let PatKind::Range(ref lhs, ref rhs, _) = pat.node,
                     let Ok(lhs) = constcx.eval(lhs, ExprTypeChecked),
                     let Ok(rhs) = constcx.eval(rhs, ExprTypeChecked)
                 ], {

--- a/clippy_lints/src/utils/hir.rs
+++ b/clippy_lints/src/utils/hir.rs
@@ -160,8 +160,8 @@ impl<'a, 'tcx: 'a> SpanlessEq<'a, 'tcx> {
             (&PatKind::Tuple(ref l, ls), &PatKind::Tuple(ref r, rs)) => {
                 ls == rs && over(l, r, |l, r| self.eq_pat(l, r))
             },
-            (&PatKind::Range(ref ls, ref le), &PatKind::Range(ref rs, ref re)) => {
-                self.eq_expr(ls, rs) && self.eq_expr(le, re)
+            (&PatKind::Range(ref ls, ref le, ref li), &PatKind::Range(ref rs, ref re, ref ri)) => {
+                self.eq_expr(ls, rs) && self.eq_expr(le, re) && (*li == *ri)
             },
             (&PatKind::Ref(ref le, ref lm), &PatKind::Ref(ref re, ref rm)) => lm == rm && self.eq_pat(le, re),
             (&PatKind::Slice(ref ls, ref li, ref le), &PatKind::Slice(ref rs, ref ri, ref re)) => {

--- a/clippy_lints/src/utils/inspector.rs
+++ b/clippy_lints/src/utils/inspector.rs
@@ -475,10 +475,14 @@ fn print_pat(cx: &LateContext, pat: &hir::Pat, indent: usize) {
             println!("{}Lit", ind);
             print_expr(cx, e, indent + 1);
         },
-        hir::PatKind::Range(ref l, ref r) => {
+        hir::PatKind::Range(ref l, ref r, ref range_end) => {
             println!("{}Range", ind);
             print_expr(cx, l, indent + 1);
             print_expr(cx, r, indent + 1);
+            match *range_end {
+                hir::RangeEnd::Included => println!("{} end included", ind),
+                hir::RangeEnd::Excluded => println!("{} end excluded", ind),
+            }
         },
         hir::PatKind::Slice(ref first_pats, ref range, ref last_pats) => {
             println!("{}Slice [a, b, ..i, y, z]", ind);

